tiledb/sm/filesystem/vfs.cc

```diff
@@ -833,18 +805,18 @@ Status VFS::move_dir(const URI& old_uri, const URI& new_uri) {
   throw UnsupportedURI(old_uri.to_string() + ", " + new_uri.to_string());
 }
 
-Status VFS::copy_file(const URI& old_uri, const URI& new_uri) {
+void VFS::copy_file(const URI& old_uri, const URI& new_uri) const {
   auto instrument = make_log_duration_instrument(old_uri, new_uri);
   // If new_uri exists, delete it or raise an error based on `force`
-  bool is_file;
-  RETURN_NOT_OK(this->is_file(new_uri, &is_file));
-  if (is_file)
-    RETURN_NOT_OK(remove_file(new_uri));
+  if (this->is_file(new_uri)) {
+    remove_file(new_uri);
+  }
```

Comment:
Can you move this to `Posix`? IIRC the object storage backends will overwrite the file themselves, and on Windows the destination file will be removed only for `Win::copy_file` to throw because it is not implemented.

Reply:
Will address in followup.